opds_import.py

```diff
@@ -356,86 +354,40 @@ def import_edition_from_metadata(
-            # not to bother with books that appeared before a
-            # certain date, and this book did in fact appear
-            # before that date. There's no reason to do anything.
-            return None
-
-        if license_pool is None:
-            # Without a LicensePool, we can't circulationdata.apply and can't create a Work.
-            error_message = "No LicensePool created for CirculationData %r, will not attempt to continue to create Work." % circulationdata
-            self.log.warn(error_message)
-            raise DBImportException(error_message)
-            return None
-
-        policy = ReplacementPolicy(
-            subjects=True,
-            links=True,
-            contributions=True,
-            rights=True,
-            link_content=True,
-            even_if_not_apparently_updated=True,
-            mirror=self.mirror,
-            http_get=self.http_get,
-        )
-        circulationdata.apply(
-            license_pool, replace=policy
-        )
-
-        return license_pool
-
 
-    def import_work(
-            self, license_pool, even_if_no_author, immediately_presentation_ready
-    ):
-        """ Making work is not based on pool's cutoff logic, but based on whether have a pool and and edition, period.
-        Don't mind if they're new or not.
-        """
-
-        # Note: pool.calculate_work will call self.set_presentation_edition(), 
-        # which will find editions attached to same Identifier.
-        # If the pool has no edition, pool.calculate_work is responsible for not allowing the work to be created.
-        work = None
-        work, is_new_work = license_pool.calculate_work(even_if_no_author=even_if_no_author)
-        # Note: if pool.calculate_work found or made a work, it already called work.calculate_presentation()
-        if (work):
-            if immediately_presentation_ready:
-                # We want this book to be presentation-ready
-                # immediately upon import. As long as no crucial
-                # information is missing (like language or title),
-                # this will do it.
-                work.set_presentation_ready_based_on_content()
-        return license_pool, work
+    @classmethod
+    def extract_last_update_dates(cls, feed):
+        parsed_feed = feedparser.parse(feed)
+        return [
+            cls.last_update_date_for_feedparser_entry(entry)
+            for entry in parsed_feed['entries']
+        ]
 
 
     def extract_feed_data(self, feed):
         """Turn an OPDS feed into lists of Metadata and CirculationData objects, 
         with associated messages and next_links.
         """
-        fp_data_meta, fp_data_circ, status_messages, next_links = self.extract_data_from_feedparser(feed=feed, data_source=self.data_source_name)
+        data_source = DataSource.lookup(self._db, self.data_source_name)
+        fp_metadata, fp_status_messages = self.extract_data_from_feedparser(feed=feed, data_source=data_source)
```

Comment:
This now checks if the data source offers licenses, and doesn't add circulation data to the metadata if not.